### PR TITLE
Fix an issue when displaying toasts for build failures

### PR DESCRIPTION
**Pull request by nickpape-msft**
Fixes https://github.com/SharePoint/sp-dev-fx-webparts/issues/297 and https://github.com/SharePoint/sp-dev-docs/issues/912

### CODE REVIEW

File: core-build/gulp-core-build/src/index.ts [hunk 1]
```diff
@@ -230,14 +230,14 @@ export function watch(watchMatch: string | string[], taskExecutable: IExecutable
                 }
                 return _finalizeWatch();
               })
-              .catch((error) => {
+              .catch((error: Error | string) => {
```
Comment (qz2017) [inline core-build/gulp-core-build/src/index.ts]: >| string [](start = 35, length = 8)

Can this one ever be string type?

Comment (pgonzal) [reply core-build/gulp-core-build/src/index.ts]: +1

Our coding standard is to always throw subclasses of Error.  If someone throws a plain string, that is a bug that should be fixed immediately.

If an OSS library is throwing these bad objects, we should have special code to wrap it, with a comment explaining why it's needed.

---
In reply to: [147839652](https://github.com/Microsoft/web-build-tools/pull/419#discussion_r147839652) [](ancestors = 147839652)

Comment (pgonzal) [reply core-build/gulp-core-build/src/index.ts]: I have proactively approved this PR, but please fix this issue (and any other occurrences of it) before merging

---
In reply to: [147844044](https://github.com/Microsoft/web-build-tools/pull/419#discussion_r147844044) [](ancestors = 147844044,147839652)

Comment (nickpape-msft) [reply core-build/gulp-core-build/src/index.ts]: I double checked and it looks like it cannot be a string type. It is normalized in the base GulpTask, and serial and parallel properly return Error objects

**Pull request merged**